src/main/java/school/lemon/changerequest/java/introduction/hw2/Task4.java

```diff
@@ -2,6 +2,11 @@
 
 public class Task4 {
     public static String charAt(String s, int i) {
-        return "";
+        int realCharAt=0;
+        if (i>=0&& s.length()>=i){realCharAt=i;}
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#floorMod-int-int-
You can use this for your purpose.